src/protocols/fp-car.ts

```diff
@@ -0,0 +1,137 @@
+// import { Result } from "@adviser/cement";
+import { FPEnvelopeTurnaround, FPEnvelopeType, FPStoreType } from "./fp-envelope";
+import { CID } from "multiformats";
+
+export interface FPGetCarReq {
+    readonly cid: CID;
+}
+/*
+ * A request to get a CAR file.
+ * As JSON, this is:
+ * {
+ *    "type": "get-car-req",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopeGetCarReq extends FPCarReq<FPGetCarReq> {
+    readonly type: FPEnvelopeType.GET_CARREQ;
+    readonly storeType: FPStoreType.CAR;
+}
+
+export interface FPGetCarRes {
+    readonly cid: CID;
+    readonly bytes: Uint8Array;
+}
+/*
+ * The response to a GetCar request.
+ * As JSON, this is:
+ * {
+ *    "type": "get-car-res",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *       "bytes": [0x65, 0x66, 0x67, ...],
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopeGetCarRes extends FPCarRes<FPGetCarRes> {
+    readonly type: FPEnvelopeType.GET_CARRES;
+    readonly storeType: FPStoreType.CAR;
+}
+
+export interface FPPutCarReq {
+    readonly cid: CID;
+    readonly bytes: Uint8Array;
+}
+/*
+ * A request to get a CAR file.
+ * As JSON, this is:
+ * {
+ *    "type": "put-car-req",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *       "bytes": [0x65, 0x66, 0x67, ...],
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopePutCarReq extends FPCarReq<FPPutCarReq> {
+    readonly type: FPEnvelopeType.PUT_CARREQ;
+    readonly storeType: FPStoreType.CAR;
+}
+
+export interface FPDelCarReq {
+    readonly cid: CID;
+}
+/*
+ * The response to a PutCar request.
```

Comment:
the signed upload url can be returned as a redirect, but today we put it in a json envelope, either is ok, but the result here should just be a URL in the s3 protocol